### ISSUE TITLE
Two levels of defer in EventSourcedBehavior

### ISSUE THREAD

**Issue by johanandren** [akka/akka#26494]
The `EventSourcedBehaviorImpl` is a deferred behavior, but when it is run, it returns a new deferred behavior. Could we do this only once instead?

**Comment by patriknw** [comment]
`apply` returns:
```
    Behaviors
      .supervise {
        Behaviors.setup[Command]
```

The supervise is needed for the stop/backoff of journal failures.

The purpose of the outer deferred behavior is to allow it be a builder but still an ordinary Behavior that can be spawned without an extra build step.

Not sure I understand or see the problem.

**Comment by johanandren** [comment]
Just an extra undefer that felt unnecessary but maybe it is unavoidable, not a big deal anyways so we can close.
